cypress/integration/testBorna/test.cy.js

```diff
@@ -0,0 +1,179 @@
+    var block_category_name = uniqid();
+    var slug_name = makeid(10);
+    beforeEach(() => cy.login());
+
+    it("Test creation of page builder block module using the New Block UI button if the is no category yet", () => {
+        cy.login();
+        cy.visit("/page-builder/page-blocks");
+        getPlaceholder("Search blocks").should("exist");
+        cy.contains("New Block").click();
+        cy.contains("Create a new block category").click();
+        cy.get(".mdc-text-field__input").eq(0).type(block_category_name);
+        cy.get(".mdc-text-field__input").eq(1).type(slug_name);
+        cy.get(".mdc-text-field__input").eq(2).type(block_category_name);
+        cy.contains("Save block category").click();
+        cy.visit("/page-builder/page-blocks");
+        getPlaceholder("Search blocks").should("exist");
+        cy.contains("New Block").click();
+        cy.contains(slug_name).click();
+        cy.contains("Save Changes").click();
+
+        cy.contains("New Block").should("exist");
+    });
+    /*
+	it("Test creation of page builder block module using the New Block UI button", () => {
+        cy.visit("/page-builder/page-blocks");
+		getPlaceholder("Search blocks").should("exist");
+		containsClick("New Block");
+		containsClick(slug_name);
+		containsClick("Save Changes");
+		cy.contains("New Block").should("exist");
+    });
+	*/
+
+    it("Test the functionality of editing the block category", () => {
+        cy.visit("/page-builder/block-categories");
+        getPlaceholder("Search block categories").should("exist");
+        cy.contains(block_category_name).click();
+        cy.get(".mdc-text-field__input")
+            .eq(2)
+            .type(block_category_name + "1");
+        cy.contains("Save block category").click();
+        cy.contains("New Block").should("exist");
+        cy.wait(1000);
+    });
+
+    it("Test the functionality of the duplicate block button", () => {
+        cy.visit("/page-builder/page-blocks");
+        getPlaceholder("Search blocks").should("exist");
+        cy.contains(block_category_name).click();
+        cy.get(
+            ".rmwc-icon.rmwc-icon--component.material-icons.css-8j2ujd-DuplicateButton.eglpe7i6.mdc-icon-button"
```

Comment:
Writing selectors like these is not maintainable. The moment we change the button class, this test will start failing. 

Cypress testing library will help with this, where we can basically just find a button by text. Alternatively, if even CTL doesn't help, we can add a test ID on the button, and then you'll be able to find the button by it.

Reply:
I agree that selectors like these aren't maintainable i'll try to see if I can get it some other way by using CTL, but for this particular case I think that a new test ID is needed, yes.